test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -296,6 +296,46 @@ function ExecuteWebRequest
     return $result
 }
 
+# Starts the ClientCertificateCheck listener
+function StartClientCertificateCheck 
+{
+    param([int]$Port)
+    $initTimeoutSeconds  = 15
+    $projectRootPath     = Resolve-Path (Join-Path $PSScriptRoot "\..\..\..\..\")
+    $appSrcPath          = Join-Path $projectRootPath  'test\tools\ClientCertificateCheck'
+    $appPublishPath      = Join-Path $appSrcPath 'bin'
+    $appDllPath          = Join-Path $appPublishPath 'ClientCertificateCheck.dll'
+    $serverPfxPath       = Join-Path $appSrcPath 'ServerCert.pfx'
+    $serverPfxPassword   = 'password'
+    $initCompleteMessage = 'Now listening on'
+    
+    $timeOut = (get-date).AddSeconds($initTimeoutSeconds)
+    $Job = Start-Job {
+        Push-Location $using:appPublishPath
+        dotnet $using:appDllPath $using:serverPfxPath $using:serverPfxPassword $using:port
+    }
+    # Wait until the app is running or until the initTimeoutSeconds have been reached
+    do
+    {
+        Start-Sleep -Seconds 1
+        $containerStatus = $Job.ChildJobs[0].Output | Out-String
+        $isRunning = $containerStatus -match $initCompleteMessage
+    }
+    while (-not $isRunning -and (get-date) -lt $timeOut)
+
+    if (-not $isRunning) {
+        throw 'Container did not start before the timeout was reached.'
+    }
+    return $job
+}
+
+# Stops the ClientCertificateCheck listener
+function StopClientCertificateCheck 
```

Comment:
Maybe `Stop-ClientCertificateCheck`.
If we'll use the application in multiple test files we should it once and shouldn't stop it.